concourse/pipelines/deploy-cloudfoundry.yml

```diff
@@ -159,40 +205,59 @@ jobs:
           params:
             file: terraform/cf.tfstate
 
-  - name: generate-cf-secrets
-    plan:
-      - get: paas-cf
-        passed: ['init']
-      - get: pipeline-trigger
-        passed: [ 'init' ]
-        trigger: true
-
-      - task: generate
+      - task: extract-cf-terraform-outputs
         config:
-          image: docker:///governmentpaas/mksecrets
+          platform: linux
+          image: docker:///ruby#2.2.3-slim
           inputs:
             - name: paas-cf
+            - name: cf-tfstate
           run:
             path: sh
             args:
-              - -c
-              - -e
-              - |
-                ./paas-cf/manifests/cf-manifest/scripts/generate-cf-secrets.sh > cf-secrets.yml
-                ls -l cf-secrets.yml
+            - -e
+            - -c
+            - |
+              SCPATH="./paas-cf/concourse/scripts"
+              SCFILE="extract_tf_vars_from_terraform_state.rb"
+              $SCPATH/$SCFILE < cf-tfstate/cf.tfstate > cf.tfstate.sh
+              ls -l cf.tfstate.sh
 
-      - task: upload
+      - task: init-db
         config:
-          image: docker:///governmentpaas/curl-ssl
+          image: docker:///governmentpaas/psql
           inputs:
+            - name: terraform-variables
             - name: paas-cf
-            - name: generate
+            - name: extract-cf-terraform-outputs
           run:
-            path: paas-cf/concourse/scripts/s3init.sh
+            path: sh
             args:
-              - {{state_bucket}}
-              - cf-secrets.yml
-              - generate/cf-secrets.yml
+            - -e
+            - -c
+            - |
+              . terraform-variables/cf-secrets.tfvars.sh
+              . extract-cf-terraform-outputs/cf.tfstate.sh
+              export PGPASSWORD=${TF_VAR_secrets_cf_db_master_password}
+              connect_str="-h ${TF_VAR_cf_db_address} -U dbadmin"
+
+              # Create roles
+              psql ${connect_str} -d postgres -c "SELECT rolname FROM pg_roles WHERE rolname = 'api'" \
+                | grep -q 'api' || psql ${connect_str} -d postgres \
+                -c "CREATE USER api WITH PASSWORD '${TF_VAR_secrets_cf_db_api_password}' ROLE dbadmin"
+              psql ${connect_str} -d postgres -c "SELECT rolname FROM pg_roles WHERE rolname = 'uaa'" \
+                | grep -q 'uaa' || psql ${connect_str} -d postgres \
+                -c "CREATE USER uaa WITH PASSWORD '${TF_VAR_secrets_cf_db_uaa_password}' ROLE dbadmin"
+
+              for db in api uaa; do
+                # Create database
+                psql ${connect_str} -d postgres -l | grep -q " ${db} " || \
+                  psql ${connect_str} -d postgres -c "CREATE DATABASE ${db} OWNER ${db}"
+                # Enable extensions
+                for ext in citext pgcrypto pg_stat_statements; do
+                  psql ${connect_str} -d ${db} -c "CREATE EXTENSION IF NOT EXISTS ${ext}"
+                done
+              done
```

Comment:
I think this is long enough to merit being pulled out into a separate script.  That way it'll get shellchecked etc..